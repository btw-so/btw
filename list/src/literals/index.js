import { keyMirror } from "@gilbarbara/helpers";

// import { Status } from "types";

export const ActionTypes = keyMirror({
  GITHUB_GET_REPOS_REQUEST: undefined,
  GITHUB_GET_REPOS_SUCCESS: undefined,
  GITHUB_GET_REPOS_FAILURE: undefined,
  HIDE_ALERT: undefined,
  SHOW_ALERT: undefined,
  SET_APP_OPTIONS: undefined,
  USER_LOGIN_REQUEST: undefined,
  USER_LOGIN_SUCCESS: undefined,
  USER_LOGIN_FAILURE: undefined,
  USER_LOGOUT_REQUEST: undefined,
  USER_LOGOUT_SUCCESS: undefined,
  USER_LOGOUT_FAILURE: undefined,
  GET_USER: undefined,
  GET_USER_SUCCESS: undefined,
  GET_USER_FAILURE: undefined,
  GENERATE_OTP: undefined,
  GENERATE_OTP_SUCCESS: undefined,
  GENERATE_OTP_FAILURE: undefined,
  VERIFY_OTP: undefined,
  VERIFY_OTP_SUCCESS: undefined,
  VERIFY_OTP_FAILURE: undefined,
  UPDATE_USER: undefined,
  UPDATE_USER_SUCCESS: undefined,
  UPDATE_USER_FAILURE: undefined,
  RESET_USER_STATE: undefined,
  GET_LIST: undefined,
  GET_LIST_SUCCESS: undefined,
  GET_LIST_FAILURE: undefined,
  NEW_LIST_NODE: undefined,
  EDIT_LIST_NODE_POS: undefined,
  EDIT_LIST_NODE_CONTENT: undefined,
  UPSERT_LIST_NODE: undefined,
  CHANGE_SELECTED_NODE: undefined,
  BATCH_PUSH_NODES: undefined,
  SAVE_NOTE_CONTENT: undefined,
  GET_NOTE: undefined,
  GET_NOTE_SUCCESS: undefined,
  GET_NOTE_FAILURE: undefined,
  ADD_FILE: undefined,
  ADD_FILE_SUCCESS: undefined,
  ADD_FILE_FAILURE: undefined,
  GET_FILE: undefined,
  GET_FILE_SUCCESS: undefined,
  GET_FILE_FAILURE: undefined,
  GET_PINNED_NODES: undefined,
  GET_PINNED_NODES_SUCCESS: undefined,
  GET_PINNED_NODES_FAILURE: undefined,
  GET_PUBLIC_NOTE: undefined,
  GET_PUBLIC_NOTE_SUCCESS: undefined,
  GET_PUBLIC_NOTE_FAILURE: undefined,
  SEARCH_NODES: undefined,
  SEARCH_NODES_SUCCESS: undefined,
  SEARCH_NODES_FAILURE: undefined,
});

export const STATUS = {
  IDLE: "idle",
  RUNNING: "running",
  READY: "ready",
  SUCCESS: "success",
  ERROR: "error",
  PINNED: "pinned",
};
