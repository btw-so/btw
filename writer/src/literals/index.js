import { keyMirror } from "@gilbarbara/helpers";

// import { Status } from "types";

export const ActionTypes = keyMirror({
  GITHUB_GET_REPOS_REQUEST: undefined,
  GITHUB_GET_REPOS_SUCCESS: undefined,
  GITHUB_GET_REPOS_FAILURE: undefined,
  HIDE_ALERT: undefined,
  SHOW_ALERT: undefined,
  SET_APP_OPTIONS: undefined,
  USER_LOGIN_REQUEST: undefined,
  USER_LOGIN_SUCCESS: undefined,
  USER_LOGIN_FAILURE: undefined,
  USER_LOGOUT_REQUEST: undefined,
  USER_LOGOUT_SUCCESS: undefined,
  USER_LOGOUT_FAILURE: undefined,
  GET_USER: undefined,
  GET_USER_SUCCESS: undefined,
  GET_USER_FAILURE: undefined,
  GENERATE_OTP: undefined,
  GENERATE_OTP_SUCCESS: undefined,
  GENERATE_OTP_FAILURE: undefined,
  VERIFY_OTP: undefined,
  VERIFY_OTP_SUCCESS: undefined,
  VERIFY_OTP_FAILURE: undefined,
  UPDATE_USER: undefined,
  UPDATE_USER_SUCCESS: undefined,
  UPDATE_USER_FAILURE: undefined,
  GET_NOTES: undefined,
  GET_NOTES_SUCCESS: undefined,
  GET_NOTES_FAILURE: undefined,
  UPSERT_NOTE: undefined,
  UPSERT_NOTE_SUCCESS: undefined,
  UPSERT_NOTE_FAILURE: undefined,
  CREATE_NEW_NOTE: undefined,
  SELECT_NOTE: undefined,
  SAVE_NOTE_CONTENT: undefined,
  IMPORT_NOTES: undefined,
  ADD_CUSTOM_DOMAIN: undefined,
  ADD_CUSTOM_DOMAIN_SUCCESS: undefined,
  ADD_CUSTOM_DOMAIN_FAILURE: undefined,
  PUBLISH_NOTE: undefined,
  PUBLISH_NOTE_SUCCESS: undefined,
  PUBLISH_NOTE_FAILURE: undefined,
  ARCHIVE_NOTE: undefined,
  ARCHIVE_NOTE_SUCCESS: undefined,
  ARCHIVE_NOTE_FAILURE: undefined,
  DELETE_NOTE: undefined,
  DELETE_NOTE_SUCCESS: undefined,
  DELETE_NOTE_FAILURE: undefined,
  GET_NOTE: undefined,
  GET_NOTE_SUCCESS: undefined,
  GET_NOTE_FAILURE: undefined,
  RESET_USER_STATE: undefined,
  SET_NOTE_SLUG: undefined,
  SET_NOTE_SLUG_SUCCESS: undefined,
  SET_NOTE_SLUG_FAILURE: undefined,
  SEARCH_NOTES: undefined,
  SEARCH_NOTES_SUCCESS: undefined,
  SEARCH_NOTES_FAILURE: undefined,
});

export const STATUS = {
  IDLE: "idle",
  RUNNING: "running",
  READY: "ready",
  SUCCESS: "success",
  ERROR: "error",
};
