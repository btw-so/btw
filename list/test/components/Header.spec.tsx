import React from 'react';

import { ActionTypes } from 'literals';

import Header from 'components/Header';

import { fireEvent, render, screen } from 'test-utils';

const mockDispatch = jest.fn();

describe('Header', () => {
  it('should render properly', () => {
    render(<Header />);

    expect(screen.getByRole('banner')).toMatchSnapshot();
  });

  it('should handle clicks', async () => {
    render(<Header />, { mockDispatch });
    fireEvent.click(screen.getByRole('button')!);

    expect(mockDispatch).toHaveBeenCalledWith({
      type: ActionTypes.USER_LOGOUT_REQUEST,
    });
  });
});
