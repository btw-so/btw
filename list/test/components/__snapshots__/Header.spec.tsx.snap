// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
.c1 {
  box-sizing: border-box;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  position: relative;
  width: 100%;
}

.c5 {
  display: inline-block;
  line-height: 0;
}

.c5 svg {
  height: auto;
  max-height: 100%;
  width: 16px;
}

.c3 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 0;
}

.c3 svg {
  height: 4.2rem;
  max-height: 100%;
  width: auto;
}

.c0 {
  background-color: #113740;
  height: 70px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 200;
}

.c0:before {
  background-color: #00b4d5;
  bottom: 0;
  content: '';
  height: 0.2rem;
  left: 0;
  position: absolute;
  right: 0;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 8px;
  padding-top: 8px;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.3rem;
  padding: 8px;
}

.c4.active {
  color: #fff;
}

.c4 span {
  display: inline-block;
  margin-right: 0.4rem;
  text-transform: uppercase;
}

@media screen and (min-width:1024px) {
  .c1 {
    padding-left: 32px;
    padding-right: 32px;
  }
}

@media screen and (min-width:1024px) {
  .c4 {
    font-size: 1.6rem;
  }
}

<header
  class="c0"
  data-testid="Header"
>
  <div
    class="c1 c2"
  >
    <div
      class="c3"
    >
      <svg
        src="/media/brand/icon.svg"
      />
    </div>
    <button
      class="c4"
    >
      <span>
        logout
      </span>
      <svg
        class="c5"
        height="100%"
        src="/media/icons/sign-out.svg"
        width="16px"
      />
    </button>
  </div>
</header>
`;
